great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -2437,6 +2440,21 @@ def _get_metric_configuration_tuples(
 
         return metric_configurations_list
 
+    @classmethod
+    def validate_config(
+        cls, project_config: Union[DataContextConfig, Mapping]
```

Comment:
`validate_config()` 's behavior has been extended to always return a `DataContextConfig` or error out.  This helps with `mypy` errors, which were complaining that parameters were being `Union[DataContextConfig, Mapping]` in one place and `DataContextConfig` in another. 

It has also been moved up to `AbstractDataContext` because `BaseDataContext` and `CloudDataContext` both use it. 


This is the previous implementation
```python
    @classmethod
    def validate_config(cls, project_config: Union[DataContextConfig, Mapping]) -> bool:
        if isinstance(project_config, DataContextConfig):
            return True
        try:
            dataContextConfigSchema.load(project_config)
        except ValidationError:
            raise
        return True
```

Reply:
Is this name still appropriate? Is there perhaps a better name to signal what it is doing now that it isn't returning a `bool`?